graphql_compiler/interpreter/typedefs.py

```diff
@@ -172,8 +172,84 @@ def project_property(
-        # TODO(predrag): Add more docs in an upcoming PR.
+        """Produce the values for a given property for each of an iterable of input DataTokens.
+
+        In situations such as outputting property values or applying filters to properties,
+        the interpreter needs to get the value of some property field for a series of DataTokens.
+
+        For example, consider the following query:
+            {
+                Foo {
+                    bar @output(out_name: "bar_value")
+                }
+            }
+
+        Once the interpreter has used the get_tokens_of_type() function to obtain
+        an iterable of DataTokens for the Foo type, it will automatically wrap each of them in
+        a "bookkeeping" object called DataContext. These DataContext objects allow
+        the interpreter to keep track of "which data came from where"; only the DataToken value
+        within their current_token attribute is relevant to the InterpreterAdapter API.
+
+        Having obtained an iterable of DataTokens and converted it to an iterable of DataContexts,
+        the interpreter needs to get the value of the "bar" property for the tokens within
+        the contexts. To do so, the interpreter calls project_property() with the iterable
+        of DataContexts, setting current_type_name = "Foo" and field_name = "bar", requesting
+        the "bar" property's value for each DataContext with its corresponding current_token.
+        If the DataContext's current_token attribute is set to None (which may happen
+        when @optional edges are used), the property's value is considered to be None.
+
+        A simple example implementation is as follows:
+            def project_property(
+                self,
+                data_contexts: Iterable[DataContext[DataToken]],
+                current_type_name: str,
+                field_name: str,
+                **hints: Any,
+            ) -> Iterable[Tuple[DataContext[DataToken], Any]]:
+                for data_context in data_contexts:
+                    current_token = data_context.current_token
+                    property_value: Any
+                    if current_token is None:
+                        # Evaluating an @optional scope where the optional edge didn't exist.
+                        # There is no value for the named property here.
+                        property_value = None
+                    else:
+                        property_value = < load the field_name property's value for current_token >
+
+                    # Remember to always yield the DataContext alongside the produced value
+                    yield data_context, property_value
+
+        Args:
+            data_contexts: iterable of DataContext objects which specify the DataTokens whose
+                           property data needs to be loaded
+            current_type_name: name of the vertex type whose property is being loaded. Guaranteed
```

Comment:
```suggestion
            current_type_name: name of the vertex type whose property needs to be loaded. Guaranteed
```
Unless there's a reason to use different phrasing here, making it concordant with the former and subsequent args' descriptions will reduce an opportunity for confusion.

Reply:
Good catch!